core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+public class BlobMetadata {
+  private final String type;
+  private final Set<Integer> columns;
+  private final long offset;
+  private final long length;
+  private final String compressionCodec;
+
+  public BlobMetadata(String type, Set<Integer> columns, long offset, long length, @Nullable String compressionCodec) {
+    this.type = Preconditions.checkNotNull(type, "type is null");
+    this.columns = ImmutableSet.copyOf(Preconditions.checkNotNull(columns, "columns is null"));
```

Comment:
I think columns should probably be a list rather than a set because order matters when computing inputs to blobs. For example, if you have a blob for NDV across two columns you'd need to know how to construct the value tuple that is used as input to the theta sketch.

Reply:
Good catch. In fact, blob input should also allow duplicate column inputs too